roadmap/propagate-rust-panic-through-native-frame.md

```diff
@@ -0,0 +1,44 @@
+# Propagate Rust panic through native frames
+
+## Summary
+
+These roadmap items refer to when a panic is raised in a Rust function
+that is defined with `"C unwind"` ABI, as would happen if C or C++
+code invoked the following function:
+
+```rust
+extern "C unwind" fn example() {
+    panic!("Uh oh");
+}
+```
+
+In this case, the stack would look like one of the two scenarios:
+
+```
+     ...
+      |
+[Native frame]
+      |                           ^
+      | (calls)                   | (unwinding
+      v                           |  goes this
+[Rust function `example`]         |  way)
+      |                           |
+      +--- rust function panics --+
+```
+
+We can distinguish two cases:
+
+* The native frame has **no destructors**. An example of this might be a C function.
+    * The precise meaning of "no destructors" is defined in terms of the native ABI.
+    * We need to define how the Rust panic propagates across such frames.
+* The native frame **has destructors** it would like to execute. These could come from C++
+  RAII, Java finally blocks, etc.
+    * We need to define how the Rust panic is defined and which destructors would execute.
```

Comment:
It might be worth distinguishing a third case:

>* The native frame has **no destructors or any affordances for unwinding**. An example of this might be a C function compiled with `-fno-exceptions` or `-fno-unwind-tables`.
>    * The precise meaning of "affordances for unwinding" is defined in terms of the native ABI.

Reply:
ah, very nice